archive/2025-09-19-shadowleak-a-zero-click-service-side-attack-exfiltrating-sensitive-data-using-chatgpts-agent.md

```diff
@@ -0,0 +1,447 @@
+title: 'ShadowLeak: A Zero-Click, Service-Side Attack Exfiltrating Sensitive Data
+  Using ChatGPT’s Agent'
+---
+
+﻿
+
+# ShadowLeak: A Zero-Click, Service-Side Attack Exfiltrating Sensitive Data Using ChatGPT’s Deep Research Agent
+
+* * *
+
+By
+[Co-Lead Researchers: Zvika Babo, Gabi Nakibly; Contributor: Maor Uziel](https://www.radware.com/Blog/Authors/Co-Lead-Researchers-Zvika-Babo,-Gabi-Nakibly;-Contributor-Maor-Uziel)September 18, 2025
+
+### Key Insights:
+
+- We found a **zero-click** flaw in ChatGPT's **Deep Research** agent when connected to Gmail and browsing: A single crafted email quietly makes the agent leak sensitive inbox data to an attacker **with no user action or visible UI**.
+- **Service-Side Exfiltration:** Unlike prior research that relied on client-side image rendering to trigger the leak, this attack leaks data directly from OpenAI’s cloud infrastructure, making it invisible to local or enterprise defenses.
+- The attack utilizes an indirect prompt injection that can be **hidden in email HTML** (tiny fonts, white-on-white text, layout tricks) so the user never notices the commands, but the agent still reads and obeys them.
+- Well-crafted **social engineering** tricks bypassed the agent’s safety-trained restrictions, enabling the attack to succeed with a 100% success rate.
+
+![ShadowLeak: A Zero-Click, Service-Side Attack Exfiltrating Sensitive Data Using ChatGPT’s Deep Research Agent. Blog image](https://www.radware.com/RadwareSite/MediaLibraries/WordPressImages/uploads/2025/09/blog_shadow-leak_intro.jpg)
+
+## Background
+
+### What is ChatGPT’s Deep Research Agent?
+
+Deep Research is ChatGPT's powerful autonomous research mode, launched in February 2025. Imagine you could ask ChatGPT to browse the internet for you to find the most up-to-date info on something and give you a detailed report. That's exactly what Deep Research does. You give it a topic, and it spends the next five to 30 minutes digging through websites, articles and even PDFs to learn everything it can. Afterwards, it hands you a detailed, organized report of what it found. It even shows you its sources, like the website links it used, so you can check the facts for yourself. The best part? You can integrate it with applications like GitHub and Gmail, allowing it to securely perform deep dives into your personal data and documents.
+
+Deep Research can be activated by pressing the plus sign on ChatGPT’s text box and choosing “Deep research" (see below).
+
+![Deep Research](https://www.radware.com/RadwareSite/MediaLibraries/WordPressImages/uploads/2025/09/blog_shadow-leak_deep-research.jpg)
+
+Now, let’s assume you want to utilize Deep Research to generate a report based on emails in Gmail and sources from the web. In such a case the user should enable the agent to access their **Gmail** and **Web search**.
+
+A normal request for Deep Research might include:
+
+_“Do deep research on my emails from today … collect everything about …”_
```

Comment:
The "collect everything about" reduces the bar for the injection to work. We spent some time going around these specific terms with AgentFlayer. After fiddling around, you can get the injection to work without it.